client.go

```diff
@@ -428,6 +440,19 @@ func (c *Client) handleMessage(msgType int, data []byte) error {
 				return err
 			}
 			c.sigChainBlockHash = sigChainBlockHash
+
+		// client auth
+		case authChallengeAction:
+			var challenge string
+			if err := json.Unmarshal(*msg["Challenge"], &challenge); err != nil {
+				return err
+			}
+			byteChallenge, _ := hex.DecodeString(challenge)
+			signature, err := crypto.Sign(c.account.PrivateKey, byteChallenge)
```

Comment:
It is NOT safe to sign server provided data as it is. You need to make sure `byteChallenge` is NEVER a valid transaction digest.

Reply:
fixed. Add client salt to node challenge and sign on it.